openedx/core/djangoapps/password_policy/hibp.py

```diff
@@ -47,6 +50,10 @@ def range(password):
                 "7ecd77ecd77ecd7": 12,
             }
         """
+        is_encrypted = PwnedPasswordsAPI.is_sha1(password)
+        if not is_encrypted:
+            password = hashlib.sha1(password.encode('utf-8')).hexdigest().upper()
```

Comment:
Can we simply log an error and return if the provided password is not a sha1 hash to make sure we only accept the hash?

Reply:
Sorry for not adding the ACs previously, just updated the description, it was an understanding that the `range` function will accept the real password as well.